Applications/Spire/Source/Ui/UnsignedQuantityModel.cpp

```diff
@@ -0,0 +1,55 @@
+#include "Spire/Ui/UnsignedQuantityModel.hpp"
+
+using namespace boost;
+using namespace boost::signals2;
+using namespace Nexus;
+using namespace Spire;
+
+UnsignedQuantityModel::UnsignedQuantityModel(
+  std::shared_ptr<ScalarValueModel<optional<Quantity>>> model)
+  : m_model(std::move(model)),
+    m_current_connection(m_model->connect_current_signal(
+      [=] (const auto& current) {
+        on_current(current);
+      })) {}
+
+optional<Quantity> UnsignedQuantityModel::get_minimum() const {
+  auto minimum = m_model->get_minimum();
+  if(!minimum || (minimum && *minimum < 0)) {
+    return 0;
+  }
+  return minimum;
+}
+
+optional<Quantity> UnsignedQuantityModel::get_maximum() const {
+  return m_model->get_maximum();
```

Comment:
Handle the case where the maximum is negative (in which case this is models an empty quantity).